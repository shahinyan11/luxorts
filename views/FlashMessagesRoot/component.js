import FlashMessage from './FlashMessage';
import useContainer from './hook';

const FlashMessagesRoot = () => {
  const { messages } = useContainer();

  return (
    <>
      {Object.keys(messages).map((key) => {
        const message = messages[key];

        return <FlashMessage {...message} key={message.id} />;
      })}
    </>
  );
};

export default FlashMessagesRoot;
