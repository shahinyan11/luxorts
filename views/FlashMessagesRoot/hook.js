import { useSelector } from 'react-redux';
import { prop } from 'ramda';

function useContainer() {
  const messages = useSelector(prop('flash-messages'));

  return {
    messages,
  };
}

export default useContainer;
