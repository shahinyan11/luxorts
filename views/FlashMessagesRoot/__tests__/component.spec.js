import { shallow } from 'enzyme';

import FlashMessagesRoot from '../component';

const mockedHookData = {
  messages: { 1: { id: '1', duration: 3, description: {}, messageType: 'success' } },
};
jest.mock('../hook', () => jest.fn(() => mockedHookData));

describe('FlashMessagesRoot component', () => {
  const component = shallow(<FlashMessagesRoot />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
