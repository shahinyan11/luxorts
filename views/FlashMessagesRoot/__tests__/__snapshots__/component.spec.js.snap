// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashMessagesRoot component matches snapshot 1`] = `
<Fragment>
  <FlashMessage
    description={Object {}}
    duration={3}
    id="1"
    key="1"
    messageType="success"
  />
</Fragment>
`;
